samza-sql/src/main/java/org/apache/samza/sql/translator/JoinTranslator.java

```diff
@@ -307,19 +334,26 @@ private Table loadLocalTable(boolean isTablePosOnRight, List<Integer> tableKeyId
     Table<KV<SamzaSqlRelRecord, SamzaSqlRelMessage>> table =
         context.getStreamAppDescriptor().getTable(sourceTableConfig.getTableDescriptor().get());
 
-    SamzaSqlRelRecordSerdeFactory.SamzaSqlRelRecordSerde keySerde =
-        (SamzaSqlRelRecordSerdeFactory.SamzaSqlRelRecordSerde) new SamzaSqlRelRecordSerdeFactory().getSerde(null, null);
-    SamzaSqlRelMessageSerdeFactory.SamzaSqlRelMessageSerde valueSerde =
-        (SamzaSqlRelMessageSerdeFactory.SamzaSqlRelMessageSerde) new SamzaSqlRelMessageSerdeFactory().getSerde(null, null);
-    // Let's always repartition by the join fields as key before sending the key and value to the table.
-    // We need to repartition the stream denoted as table to ensure that both the stream and table that are joined
-    // have the same partitioning scheme and partition key.
-    relOutputStream
-        .partitionBy(m -> createSamzaSqlCompositeKey(m, tableKeyIds),
-            m -> m,
-            KVSerde.of(keySerde, valueSerde),
-            intermediateStreamPrefix + "table_" + joinId)
-        .sendTo(table);
+    // If local table, load the table
+    if (!isRemoteTable) {
```

Comment:
There are quite a few if else's in JoinTranslator which is one of the code smells. Can you look at a better way to factor this code?

Reply:
I refactored some part of the code.

Reply:
I think I found a better way to refactor this code. Let me refactor JoinTranslator but please take a look at responses for other comments.

Reply:
Refactored the code. I will refactor the recursion part of the code to get input types as part of subquery support.